pkg/objstore/s3/s3.go

```diff
@@ -53,13 +56,28 @@ type Config struct {
 	AccessKey       string            `yaml:"access_key"`
 	Insecure        bool              `yaml:"insecure"`
 	SignatureV2     bool              `yaml:"signature_version2"`
-	SSEEncryption   bool              `yaml:"encrypt_sse"`
 	SecretKey       string            `yaml:"secret_key"`
 	PutUserMetadata map[string]string `yaml:"put_user_metadata"`
 	HTTPConfig      HTTPConfig        `yaml:"http_config"`
 	TraceConfig     TraceConfig       `yaml:"trace"`
 	// PartSize used for multipart upload. Only used if uploaded object size is known and larger than configured PartSize.
 	PartSize uint64 `yaml:"part_size"`
+
+	// SSE config items are used for configuring server-side encryption.
+	//	encrypt_sse - An alias for sse_s3.
+	// 	sse_s3 			- When true, enables SSE-S3. ignored if KMS or C are enabled. `encrypt_sse` is kept as a legacy alias for this.
+	//.	sse_c_key		- When set to a filepath, enables SSE-C. Overrides `sse_s3`.
+	//	sse_kms_id	- When set, enables SSE-KMS. Overrides `sse_s3`. A KMS context can be optionally set via `sse_kms_context`.
+	//	sse_kms_context  - If `sse_kms_id` is set, attaches the given context to all KMS requests.
+	//
+	// NOTE: If both sse_c_key and sse_kms_id are set, we throw an error, as it's unclear what the operator wants.
+	// See https://docs.aws.amazon.com/AmazonS3/latest/dev/ServerSideEncryptionCustomerKeys.html
+	// for more details on the SSE-S3, SSE-KMS, and SSE-C options.
+	SSEEncryptionLegacy bool              `yaml:"encrypt_sse"`
```

Comment:
I think we want something like this, but what about, since we are in struct make it more generic? The problem with this approach is that the relation if user will see ALL three of them is not really sure. 

It should be clear that you can either set `sse_s3`, `see_c` or `see_kms` if possible... (: 

Or at least let's create `SSE SSEConfig` field and put all of those in such struct. I think that would work. Also let's remove legacy field and add CHANGELOG item for removal of this. What do you think?